### PR TITLE
Forecast nb

### PR DESCRIPTION
**Pull request by int-chaos**
update flaml_forecast.ipynb 

add benchamarking and comparing flaml to alternatives

### CODE REVIEW

File: notebook/flaml_forecast.ipynb [hunk 1]
```diff
@@ -10,24 +10,26 @@
...
+    "> pip install flaml[notebook]"
    ],
    "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "source": [
-    "!pip install flaml[notebook,forecast]"
+    "!pip install flaml[notebook]"
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:32]: why remove 'forecast'?

File: notebook/flaml_forecast.ipynb [hunk 2]
```diff
@@ -47,13 +49,13 @@
...
-    "# data is given in weeks, but the task is to predict monthly, so use monthly averages instead\n",
-    "data = data['co2'].resample('MS').mean()\n",
-    "data = data.fillna(data.bfill())  # makes sure there are no missing values\n",
-    "data = data.to_frame().reset_index()\n",
+    "import statsmodels.api as sm\r\n",
+    "data = sm.datasets.co2.load_pandas()\r\n",
+    "data = data.data\r\n",
+    "# data is given in weeks, but the task is to predict monthly, so use monthly averages instead\r\n",
+    "data = data['co2'].resample('MS').mean()\r\n",
+    "data = data.fillna(data.bfill()) # makes sure there are no missing values\r\n",
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:57]: recover

File: notebook/flaml_forecast.ipynb [hunk 3]
```diff
@@ -63,14 +65,14 @@
...
-    "split_idx = num_samples - time_horizon\n",
-    "X_train = data[:split_idx]  # X_train is a dataframe with two columns: time and value\n",
-    "X_test = data[split_idx:]['index'].to_frame()  # X_test is a dataframe with dates for prediction\n",
-    "y_test = data[split_idx:]['co2']  # y_test is a series of the values corresponding to the dates for prediction"
+    "# split the data into a train dataframe and X_test and y_test dataframes, where the number of samples for test is equal to\r\n",
+    "# the number of periods the user wants to predict\r\n",
+    "num_samples = data.shape[0]\r\n",
+    "time_horizon = 12\r\n",
+    "split_idx = num_samples - time_horizon\r\n",
+    "X_train = data[:split_idx] # X_train is a dataframe with two columns: time and value\r\n",
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:73]: Recover

File: notebook/flaml_forecast.ipynb [hunk 4]
```diff
@@ -63,14 +65,14 @@
...
+    "X_test = data[split_idx:]['index'].to_frame() # X_test is a dataframe with dates for prediction\r\n",
+    "y_test = data[split_idx:]['co2'].to_frame() # y_test is a dataframe of the values corresponding to the dates for prediction"
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:75]: Recover

File: notebook/flaml_forecast.ipynb [hunk 5]
```diff
@@ -117,1103 +117,341 @@
    "cell_type": "code",
    "execution_count": 5,
    "source": [
-    "'''The main flaml automl API'''\n",
-    "automl.fit(dataframe=X_train,  # training data\n",
-    "           label=('index', 'co2'),  # For 'forecast' task, label should be a tuple of strings for timestamp and value columns\n",
-    "           **settings, \n",
-    "           period=time_horizon,  # key word argument 'period' must be included for forecast task\n",
+    "'''The main flaml automl API'''\r\n",
+    "automl.fit(dataframe=X_train, # training data\r\n",
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:121]: Recover

File: notebook/flaml_forecast.ipynb [hunk 6]
```diff
@@ -117,1103 +117,341 @@
...
+    "           label=('index', 'co2'), # For 'forecast' task, label should be a tuple of strings for timestamp and value columns\r\n",
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:122]: Recover

File: notebook/flaml_forecast.ipynb [hunk 7]
```diff
@@ -117,1103 +117,341 @@
...
+    "           **settings, \r\n",
+    "           period=time_horizon, # key word argument 'period' must be included for forecast task\r\n",
```
Comment (sonichi) [inline notebook/flaml_forecast.ipynb:124]: Recover

File: notebook/flaml_forecast.ipynb (hunk 4 above)
Comment (int-chaos) [reply notebook/flaml_forecast.ipynb:75]: Do you mean recover the end of line? or the other changes or both?

**Comment by ekzhu** [comment]
Thanks for the contribution! While the comparison between other approaches on MAPE is interesting, I think it may be more beneficial to provide a time series plot showing the predicted graph of FLAML's model versus the ground truth, and other baselines.

**Pull request closed without merging**